plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -67,81 +61,64 @@ object ScalafmtPlugin extends AutoPlugin {
   private val scalafmtDoFormatOnCompile =
     taskKey[Unit]("Format Scala source files if scalafmtOnCompile is on.")
 
-  private val scalaConfig =
-    scalafmtConfig.map { c =>
-      c.map { f =>
-          StyleCache.getStyleForFileOrError(f.toString).toEither match {
-            case Right(conf) => conf
-            case Left(configErr) =>
-              throw new MessageOnlyException(
-                configErr.msg
-              )
-          }
-        }
-        .getOrElse(ScalafmtConfig.default)
+  private val scalaConfig = {
+    scalafmtConfig.map { f =>
+      if (f.exists()) {
+        f.toPath
+      } else {
+        throw new MessageOnlyException(s"File not exists: ${f.toPath}")
+      }
     }
-  private val sbtConfig = scalaConfig.map(_.forSbt)
-
-  private def filterSource(source: File, config: ScalafmtConfig): Boolean =
-    config.project.matcher.matches(source.toString)
-  private def filterScala(source: File): Boolean =
-    source.toString.endsWith(".scala")
-  private def filterSbt(source: File): Boolean =
-    source.toString.endsWith(".sbt")
-  private def filterSc(source: File): Boolean =
-    source.toString.endsWith(".sc")
+  }
+  private val sbtConfig = scalaConfig
 
   private type Input = String
   private type Output = String
 
   private def withFormattedSources[T](
       sources: Seq[File],
-      config: ScalafmtConfig
+      config: Path,
+      log: Logger,
+      writer: PrintWriter
   )(
       onError: (File, Throwable) => T,
       onFormat: (File, Input, Output) => T
   ): Seq[Option[T]] = {
+    val reporter = new ScalafmtSbtReporter(log, writer)
+    val scalafmtInstance = Scalafmt.create(this.getClass.getClassLoader).withReporter(reporter)
     sources
-      .withFilter(filterSource(_, config))
```

Comment:
this filtering will be done by `scalafmt-dynamic`

Reply:
Nice!